lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -537,10 +536,11 @@ sub _generate_jobs {
             }
 
             # add properties from dedicated database columns to settings
-            $settings{TEST}              = $job_template->test_suite->name;
+            $settings{TEST}              = $job_template->name || $job_template->test_suite->name;
             $settings{MACHINE}           = $job_template->machine->name;
             $settings{BACKEND}           = $job_template->machine->backend;
             $settings{JOB_TEMPLATE_NAME} = $job_template->name if $job_template->name;
+            $settings{TEST_SUITE_NAME}   = $job_template->test_suite->name;
```

Comment:
Can you explain why we need all three "TEST" and "JOB_TEMPLATE_NAME" and "TEST_SUITE_NAME"?

Reply:
For example, the YAML is defined like this:
```
scenarios:
  x86_64:
    sle-12-sp5-server-dvd-x86_64: 
    - apparmor_test:
        testsuite: apparmor
        settings:
          BETA: '1' 
    - apparmor_test2:
        testsuite: apparmor
        settings:
           BETA: '2'
```

Reply:
Good that you mention it. For the second example of "START_AFTER_TEST" that should not be the test suite anymore but also the job template name. So for me this is another example why I am stating that the whole change is still too UI centric as with parameterized job template we should be changing the concept of how *every* job is created. Also "to find out" the test suite description I think we can try to read out a description from a job template and fallback to the test suite.

You are already on the right track but we have not completely followed through: In my opinion we should simply define the "TEST" based on the job template's name if it exists and rely on "TEST" everywhere else where we can. I know it's more than what we thought when you started but it just shows the general problems we still have not solved yet with job templates.

Reply:
> Good that you mention it. For the second example of "START_AFTER_TEST" that should not be the test suite anymore but also the job template name. So for me this is another example why I am stating that the whole change is still too UI centric as with parameterized job template we should be changing the concept of how _every_ job is created. Also "to find out" the test suite description I think we can try to read out a description from a job template and fallback to the test suite.

The `START_AFTER_TEST` is defined in test suite table. How could users know what the job template name when they add the test suite? So I do not think this value should be the job template name.

In my opinion, if we want to get the test suite description from a job template, it will be more complex.  If we change `TEST` to `JOB_TEMPLATE_NAME`, we do not know the real test suite name or id and how to find out the its description? 

> 
> You are already on the right track but we have not completely followed through: In my opinion we should simply define the "TEST" based on the job template's name if it exists and rely on "TEST" everywhere else where we can. I know it's more than what we thought when you started but it just shows the general problems we still have not solved yet with job templates.

Sorry, I do not understand your meaning clearly. In my code, I change the `TEST` to `JOB_TEMPLATE_NAME` when it exists, does it work? is there something I need to solve? If there is, feel free to tell me, I am very happy to do.

Reply:
While what okurz writes is a little confusing, I agree:

It's called START_AFTER_TEST not START_AFTER_TESTSUITE_NAME - so if TEST comes from yaml, START_AFTER_TEST needs to follow.

Reply:
Sorry, I do not understand your meaning clearly.  If I understand correctly, whatever the `TEST` is test suite name or job template name, in Oliver's example, the job A won't be triggered.  I do not understand this sentence `The same we could apply by saying that we specify the final "job_template_name" in START_AFTER_TEST and it could be just the same a testsuite or a parameterized job template.`   Does it mean that we should use the `job_template_name` as the START_AFTER_TEST if the job_template_name exists?

Reply:
yes. And I think Oliver was giving a bad example - that's made it so confusing.

Reply:
yes, this is what it means. Still I think we should try to update the value of TEST earlier, at the time of job creation, so that for START_AFTER_TEST we would not need to care where the value of TEST came from, if from a solely test suite based job or a parameterized job template one

Reply:
> so that for START_AFTER_TEST we would not need to care where the value of TEST came from

That would be ideal, indeed. And note that the same counts for the variables of the other dependency types.